libcob/Makefile.am

```diff
@@ -126,7 +136,7 @@ lib_lm =
 endif
 
 lib_LTLIBRARIES = libcob.la $(lib_ci) $(lib_di) $(lib_vb) $(lib_vc) \
-	$(lib_od) $(lib_oc) $(lib_db) $(lib_lm)
+	$(lib_od) $(lib_oc) $(lib_db) $(lib_lm) $(lib_jni)
```

Comment:
@GitMensch With these changes, it is not clear what the actual requirements for loading that additional `libcobjni` library at runtime are. Should it be via some additional code inserted by `codegen.c` for modules that have `CALL "Java.…" statements?

@xevor11 for the moment, I guess you could just add
```
if COB_HAVE_JNI
libcob_la_SOURCES += java.c
libcob_la_LIBADD += $(JVM_LIBS)
libcob_la_CFLAGS += $(JVM_CFLAGS)
endif
```
(or something equivalent — I'm not sure what are the limitations of what can be done in these automake files.)

Reply:
for now this can be done as noted here, later on there should be a separate libcob-java.so which is delay-loaded on the first call to a java function (just as it is with libcob-bdb.so)

Reply:
With the current design, most, if not all, function calls that are generated for each `CALL "Java.…"`, should end up being implemented in that `libcobjni.so` library. I guess there could be an intermediate `cob_load_java` in `libcob.so` itself, or something similar, to be called upon module initialization; or else a `cob_resolve_java_call` that is itself implemented in `call.c`, loading `libcobjni.so` if it is not loaded already, and then calling `cob_recolve_java` from `java.c`…

Reply:
@xevor11 As it's more about the build and autotest system I've push a new commit on top of `jni-config` with a draft approach for doing the aforementioned dynamic loading of `libcobjni`. At least that appears to be working on my setup.